src/js/script.js

```diff
@@ -1482,31 +1482,65 @@ var academicDegreeSearch = function ($) {
   /**
  * #search-query specific typeahead init, event handlers
  **/
-  var $academicsDegreeSearch = $('#academics-degree-search');
+  var $academicsDegreeSearch = $('#academics-degree-search'),
+      resultCount = 0;
 
   if ($academicsDegreeSearch.length > 0) {
 
+    var addScore = function(data) {
+      for (var d in data) {
+        var degree = data[d];
+        switch(degree.programType) {
+          case 'undergraduate-degree':
+          case 'articulated-program':
+            degree.score = 100;
+            break;
+          case 'graduate-degree':
+          case 'accelerated-program':
+            degree.score = 50;
+            break;
+          case 'certificate':
+            degree.score = 25;
+            break;
+        }
+      }
+
+      return data;
+    };
+
+    var scoreSorter = function(a, b) {
+      if (a.score < b.score) {
+        return 1;
+      }
+      if (a.score > b.score) {
+        return -1;
+      }
+      return 0;
+    };
+
     var degrees = new Bloodhound({
-      identify: function(obj) { return obj.name; },
+      identify: function(obj) { return obj.name + obj.programType; },
       datumTokenizer: Bloodhound.tokenizers.obj.whitespace('name'),
       queryTokenizer: Bloodhound.tokenizers.whitespace,
-      local: searchSuggestions
+      local: addScore(searchSuggestions),
+      sorter: scoreSorter
     });
 
-    var degreesWithDefaults = function (q, sync) {
-      if (q === '') {
-        sync(degrees.get('Bachelor', 'Graduate', 'Certificate'));
-      } else {
-        degrees.search(q, sync);
-      }
-    };
+    $('.degree-search-box, .academics-search-box').on('click', '#show-all-degrees', function () {
+      window.location = '/degree-search/?search-query=' + $('#academics-degree-search').val();
```

Comment:
In dev, it is loading the correct page in both cases. window.location treats a `/url.html` as relative to the root, not the current path.